Content.Server/Forensics/Systems/ForensicPadSystem.cs

```diff
@@ -0,0 +1,147 @@
+using System.Threading;
+using Content.Shared.Examine;
+using Content.Shared.Interaction;
+using Content.Shared.Inventory;
+using Content.Server.DoAfter;
+using Content.Server.Popups;
+using Robust.Shared.Player;
+
+namespace Content.Server.Forensics
+{
+    public sealed class ForensicPadSystem : EntitySystem
+    {
+        [Dependency] private readonly DoAfterSystem _doAfterSystem = default!;
+        [Dependency] private readonly InventorySystem _inventory = default!;
+
+        [Dependency] private readonly PopupSystem _popupSystem = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<ForensicPadComponent, ExaminedEvent>(OnExamined);
+            SubscribeLocalEvent<ForensicPadComponent, AfterInteractEvent>(OnAfterInteract);
+            SubscribeLocalEvent<TargetPadSuccessfulEvent>(OnTargetPadSuccessful);
+            SubscribeLocalEvent<PadCancelledEvent>(OnPadCancelled);
+        }
+
+        private void OnExamined(EntityUid uid, ForensicPadComponent component, ExaminedEvent args)
+        {
+            if (!args.IsInDetailsRange)
+                return;
+
+            if (!component.Used)
+            {
+                args.PushMarkup(Loc.GetString("forensic-pad-unused"));
+                return;
+            }
+
+            args.PushMarkup(Loc.GetString("forensic-pad-sample", ("sample", component.Sample)));
+        }
+
+        private void OnAfterInteract(EntityUid uid, ForensicPadComponent component, AfterInteractEvent args)
+        {
+            if (component.CancelToken != null)
+            {
+                component.CancelToken.Cancel();
+                component.CancelToken = null;
+            }
```

Comment:
In hindsight I don't like these and think it should just return if the canceltoken isn't null.